### DISCUSSION TITLE
Bucket fill doesn't work with vector lines

### DISCUSSION THREAD

**Discussion by UndeadLamb** [keiwando/toonsquid#125] [General]
Not to sound pushy. I love this app is amazing but not being able to bucket fill inside vector lines needs to be put at the top of the list of what needs to be worked on even if it's just alittle improvements over each update. Because that needs to work lol I get it takes time to program it but please don't leave that for the last thing to be fix keep the improvements of that higher up on the list of things that need to be fixed.

**Comment by StasClick** [comment]


**Reply by keiwando** [reply]
Hi! 
Based on your description it sounds like you are working with [vector stroke brushes](https://toonsquid.com/handbook/brushes/vector/#vector-stroke-brushes) (as opposed to [vector shape brushes](https://toonsquid.com/handbook/brushes/vector/#vector-shape-brushes)), in which case it is by design that unconnected vector paths don't create a filled region. The stroke thickness does not change anything about the geometry of the vectors, which is the only thing that matters when using the fill tool on vector contents. 
You can find more details on this here: [Notes on Vector Filling With Stroked Paths](https://toonsquid.com/updates/ToonSquid-2_0_17/#vector-fill-stroked-paths) 

To avoid confusion: This is also something completely different from the original post in this thread, which is about unreliabilities when filling vector shape regions that are actually closed, and not about automatic gap closing of stroked vector paths that aren't actually connected. Features such as vector stroke brushes, the ability to have open subpaths and stroked paths in vector layers, vector groups and the ability to use the fill tool with path layers, vector groups and stroked paths were not even a part of ToonSquid until the ToonSquid 2.0 update which was released two years after the original post of this thread.
